backend/capellacollab/cli/ws.py

```diff
@@ -0,0 +1,355 @@
+    "--namespace",
+    "-n",
+    help="Kubernetes namespace",
+    default_factory=get_current_namespace,
+)
+
+
+@app.command()
+def volumes(namespace: Annotated[str, NamespaceOption]):
+    """List all Persistent Volume Claims in a kubernetes namespace."""
+    core_api = client.CoreV1Api()
+
+    for item in core_api.list_namespaced_persistent_volume_claim(
+        namespace=namespace, watch=False
+    ).items:
+        print(item.metadata.name)
+
+
+@app.command()
+def ls(
+    volume_name: str,
+    namespace: Annotated[str, NamespaceOption],
+    path: str = MOUNT_PATH,
+):
+    """List all files on a path in a Kubernetes Persistent Volume."""
+    v1 = client.CoreV1Api()
+
+    with pod_for_volume(volume_name, namespace, MOUNT_PATH, v1) as pod_name:
+        for data in stream_tar_from_pod(pod_name, namespace, ["ls", path], v1):
+            sys.stdout.write(data.decode("utf-8", "replace"))
+
+
+@app.command()
+def backup(
+    volume_name: str,
+    namespace: Annotated[str, NamespaceOption],
+    out: Path = Path.cwd(),
+):
+    """Create a backup of all content in a Kubernetes Persistent Volume."""
+    v1 = client.CoreV1Api()
+    targz = out / f"{volume_name}.tar.gz"
+
+    with pod_for_volume(volume_name, namespace, MOUNT_PATH, v1) as pod_name:
+        print(f"Downloading workspace volume to '{targz}'")
+
+        with targz.open("wb") as outfile:
+            for data in stream_tar_from_pod(
+                pod_name, namespace, ["tar", "zcf", "-", MOUNT_PATH], v1
+            ):
+                outfile.write(data)
+
+
+@app.command()
+def restore(
+    volume_name: str,
+    tarfile: Annotated[Path, typer.Argument(exists=True)],
+    namespace: Annotated[str, NamespaceOption],
+    access_mode: str = "ReadWriteMany",
+    storage_class_name: str = "persistent-sessions-csi",
+    user_id: str | None = None,
+):
+    """Restore a backup to a Kubernetes Persistent Volume.
+
+    If the volume does not exist, it is created. Note that
+    `access-mode` and `storage-class-name` should match settings
+    for your cluster.
+
+    Optionally a `user-id` can be provided. All files then be owned
+    by this user id.
+    """
+    v1 = client.CoreV1Api()
+
+    create_persistent_volume(
+        volume_name, namespace, access_mode, storage_class_name, v1
+    )
+
+    with pod_for_volume(
+        volume_name, namespace, MOUNT_PATH, v1, read_only=False
+    ) as pod_name:
+        print(f"Restoring workspace volume to '{volume_name}'")
+
+        with tarfile.open("rb") as infile:
+            stream_tar_to_pod(pod_name, namespace, infile, v1)
+
+        adjust_directory_permissions(
+            pod_name,
+            namespace,
+            v1,
+            user_id,
+        )
+
+
+@contextlib.contextmanager
+def pod_for_volume(
+    volume_name: str,
+    namespace: str,
+    mount_path: str,
+    v1: client.CoreV1Api,
+    read_only=True,
+):
+    name = f"ws-download-{volume_name}-{uuid.uuid1()}"[:63]
```

Comment:
```suggestion
    name = f"ws-download-{uuid.uuid1()}"
```
I am not sure whether you investigated this problem further but when we used the cli, I think we run into some problems when there was a `-` at the last position of the name. So probably just using this here is sufficient.

Reply:
I wanted to have a container name that I can relate to the volume it's supposed to work with. Hence the volume name.

Reply:
I'd propose to use the original volume name as label of the deployment. We can still relate where it comes from and the issue mentioned by @dominik003  would be fixed.

Reply:
Changed to use only the volume name. I kept the name variable, though, for readability.